cmd/nodepool/kubevirt/create.go

```diff
@@ -2,75 +2,118 @@ package kubevirt
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strings"
 
-	"github.com/openshift/hypershift/examples/fixtures"
+	"github.com/openshift/hypershift/cmd/cluster/kubevirt/params"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	corev1 "k8s.io/api/core/v1"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/pointer"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 
 	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
 	"github.com/openshift/hypershift/cmd/nodepool/core"
 )
 
+func DefaultOptions() *KubevirtPlatformCreateOptions {
+	return &KubevirtPlatformCreateOptions{
+		KubevirtPlatformOptions: KubevirtPlatformOptions{
+			Memory:               "4Gi",
+			Cores:                2,
+			RootVolumeSize:       32,
+			AttachDefaultNetwork: pointer.Bool(true),
+		},
+		QoSClass: "Burstable",
+	}
+}
+
+func BindOptions(opts *KubevirtPlatformCreateOptions, flags *pflag.FlagSet) {
+	flags.StringVar(&opts.Memory, "memory", opts.Memory, "The amount of memory which is visible inside the Guest OS (type BinarySI, e.g. 5Gi, 100Mi)")
+	flags.Uint32Var(&opts.Cores, "cores", opts.Cores, "The number of cores inside the vmi, Must be a value greater or equal 1")
+	flags.StringVar(&opts.RootVolumeStorageClass, "root-volume-storage-class", opts.RootVolumeStorageClass, "The storage class to use for machines in the NodePool")
+	flags.Uint32Var(&opts.RootVolumeSize, "root-volume-size", opts.RootVolumeSize, "The size of the root volume for machines in the NodePool in Gi")
+	flags.StringVar(&opts.RootVolumeAccessModes, "root-volume-access-modes", opts.RootVolumeAccessModes, "The access modes of the root volume to use for machines in the NodePool (comma-delimited list)")
+	flags.StringVar(&opts.RootVolumeVolumeMode, "root-volume-volume-mode", opts.RootVolumeVolumeMode, "The volume mode of the root volume to use for machines in the NodePool. Supported values are \"Block\", \"Filesystem\"")
+	flags.StringVar(&opts.ContainerDiskImage, "containerdisk", opts.ContainerDiskImage, "A reference to docker image with the embedded disk to be used to create the machines")
```

Comment:
one oddity we have to deal with is that there are cli args exposed in the `hypershift` cli that we didn't want carried over to `hcp` cli. 

for kubevirt, the `--containerdisk` cli arg is one of those options. In our case, we haven't used that arg in our development flows for quite some time now, so it can be removed entirely. There might be other cases though where the dev cli (hypershift) and productized cli (hcp) differ slightly. I think that's part of the reason that the tools split the way they did.

Reply:
Perhaps we can use this as a good forcing function to make the decision about why we have it in `hypershift` if we don't have it in `hcp`! If it's intentional, we can factor a base set of options and then decorate each side with ancillary things as necessary. What we have now is implicit at best - what would you like me to do with these? I can delete this arg entirely, and are the others you speak of in other platforms?

Reply:
FWIW this same oddity applies to AWS as well. There are additional AWS flags in `hypershift` than in `hcp`. May be true for `Agent` platform as well.

Reply:
@bryan-cox can you speak to whether that's intended or emergent behavior?

Reply:
(a good number of the flags that differed in KubeVirt were accidental, and *should* have been in `hcp`, but were not)

Reply:
Yeah we went through an  exercise to figure out what flags from the `hypershift cli` should be supported officially in the `hcp cli`.

Reply:
> a good number of the flags that differed in KubeVirt were accidental, and should have been in hcp

From what I remember, those missing kubevirt args were between the `hcp create cluster` and `hcp create nodepool` commands. Both those commands are "productized"

I what we're talking about here in this comment section is more about the differences between `hypershift` (dev cli) and `hcp` (productized cli).  There were some dev cli args (like the container disk one for KubeVirt) that weren't meant to be productized.


we made this kind of complicated :( Speaking for the KubeVirt platform, i'm willing to drop all discrepancies that exist between the hypershift and hcp cli... meaning if it's kubevirt related and only in hypershift cli, we can drop it entirely.